mne/io/artemis123/artemis123.py

```diff
@@ -259,13 +312,137 @@ class RawArtemis123(BaseRaw):
+                                "active (%d) HPI coils are " +
+                                "not the same.")
+                        raise RuntimeError(mesg % (len(hpi_head),
+                                                   len(hpi_dev)))
+
+                    head_to_dev_t, order = _fit_dev_head_trans(hpi_dev,
+                                                               hpi_head)
+
+                    # set the device to head transform
+                    self.info['dev_head_t'] = \
+                        Transform(FIFF.FIFFV_COORD_DEVICE,
+                                  FIFF.FIFFV_COORD_HEAD, head_to_dev_t)
+
+                    # fill in hpi_results
+                    hpi_result = dict()
+
+                    # add HPI points in device coords...
+                    dig = []
+                    for idx, point in enumerate(hpi_dev):
+                        dig.append({'r': point, 'ident': idx + 1,
+                                    'kind': FIFF.FIFFV_POINT_HPI,
+                                    'coord_frame': FIFF.FIFFV_COORD_DEVICE})
+                    hpi_result['dig_points'] = dig
+
+                    # attach Transform
+                    hpi_result['coord_trans'] = self.info['dev_head_t']
+
+                    # 1 based indexing
+                    hpi_result['order'] = order + 1
+                    hpi_result['used'] = np.arange(3) + 1
+
+                    dig_dists = cdist(hpi_head, hpi_head)
+                    dev_dists = cdist(hpi_dev, hpi_dev)
+                    tmp_dists = np.abs(dig_dists - dev_dists)
+                    hpi_result['dist_limit'] = tmp_dists.max() * 1.1
+                    if hpi_result['dist_limit'] > 0.005:
+                        warn('Large difference between digitized geometry' +
+                             ' and HPI geometry. Max coil to coil difference' +
+                             ' is %0.2f cm\n' % (100. * tmp_dists.max()) +
+                             'beware of *POOR* head localization')
+
+                    hpi_result['good_limit'] = 0.98
+
+                    self.info['hpi_results'] = [hpi_result]
+
+                else:
+                    logger.info('Assuming Cardinal HPIs')
+                    nas = hpi_dev[0]
+                    lpa = hpi_dev[2]
+                    rpa = hpi_dev[1]
+                    t = get_ras_to_neuromag_trans(nas, lpa, rpa)
+                    self.info['dev_head_t'] = \
+                        Transform(FIFF.FIFFV_COORD_DEVICE,
+                                  FIFF.FIFFV_COORD_HEAD, t)
+
+                    # transform fiducial points
+                    nas = apply_trans(t, nas)
+                    lpa = apply_trans(t, lpa)
+                    rpa = apply_trans(t, rpa)
+
+                    hpi = [nas, rpa, lpa]
+                    self.info['dig'] = _make_dig_points(nasion=nas, lpa=lpa,
+                                                        rpa=rpa, hpi=hpi)
```

Comment:
This branching seems too extensive. Can't you do:
```
if pos_fname is None:
    logger.info('Assuming ...')
    ...
    info['dig'] = ...
...
```
and then have the localization parts (computing `info['hpi_results']`) conserved?